modules/codefresh/Makefile.sync

```diff
@@ -0,0 +1,46 @@
+CODEFRESH_CLI_IMAGE ?= codefresh/cli
+CODEFRESH_CLI_VERSION ?= latest
+
+
+export CODEFRESH_CLI ?= $(DOCKER_RUN) -v$(HOME)/.cfconfig:/root -v$(shell pwd):$(CODEFRESH_CLI_SHARED_DIR) $(CODEFRESH_CLI_IMAGE):$(CODEFRESH_CLI_VERSION)
+export CODEFRESH_CLI_SHARED_DIR ?= /data
+
+PIPELINES ?= $(call pipelines,$(ACCOUNT))
+
+define pipelines
+	$(value $(shell echo $(1)_PIPELINES | tr 'a-z' 'A-Z'))
+endef
+
+define codefresh-use-context
+	@$(CODEFRESH_CLI) auth use-context $(PROJECT)-$(1)
+endef
+
+## Authentificate on codefresh account
+codefresh/sync/auth/%:
+	$(call assert-set,CF_API_KEY)
+	$(call assert-set,PROJECT)
+	@$(CODEFRESH_CLI) auth create-context $(PROJECT)-$(subst codefresh/sync/auth/,,$@) --api-key $(CF_API_KEY)
```

Comment:
I think if the ENV is `CF_API_KEY` we don't need to pass it on the command line

Reply:
No. `CF_API_KEY` env overrides key for current context, but to do `auth` need to specify it implicit.